src/util.c

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <os.h>
+
+#define B58_MAX_INPUT_SIZE 120
+
+unsigned char const BASE58ALPHABET[] = {
+    '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F',
+    'G', 'H', 'J', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W',
+    'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'm',
+    'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
+
+void bin2hex(uint8_t *dst, uint8_t *data, uint64_t inlen) {
+    static uint8_t const hex[] = "0123456789abcdef";
+    for (uint64_t i = 0; i < inlen; i++) {
+        dst[2*i+0] = hex[(data[i]>>4) & 0x0F];
+        dst[2*i+1] = hex[(data[i]>>0) & 0x0F];
+    }
+    dst[2*inlen] = '\0';
+}
+
+int bin2dec(uint8_t *dst, uint64_t n) {
+    if (n == 0) {
+        dst[0] = '0';
+        dst[1] = '\0';
+        return 1;
+    }
+    // determine final length
+    int len = 0;
+    for (uint64_t nn = n; nn != 0; nn /= 10) {
```

Comment:
Why not use `x`? It seems that `nn` is error prone.

Reply:
It's the same code that I got from the base project (`ledger-app-sia`), so it didn't bother me.

To be honest, it's not being used. I just kept there because it was easy and seemed like it could be useful. But now that all functionality is done and this is not used, I'll remove it.

Reply:
Removed `bin2hex` and `bin2dec`, both unused.